### DISCUSSION TITLE
Suggestion - Automation for guide.

### DISCUSSION THREAD

**Discussion by bagdydad** [YouHaveTrouble/minecraft-optimization#129] [Show and tell]
I made a tool that will automatically apply the recommended settings (and some other settings of my own choice) to your server, it's open source and you can change the settings as you wish. If this interests you, add a link, it will be useful for lazy people like me :) https://github.com/bagdydad/MCSO
(If you think this is an advertising attempt, you can remove this discussion.)

**Comment by YouHaveTrouble** [comment]
This is a terrible idea. The entire point of the guide is to explain and educate, not to provide exact values to set. What you call "recommended settings" are not recommended. They are starting points to adjust based on the servers needs. There are no magical "best" settings that will work for every server and every server owner. That's a flawed view point.

Automatically setting configs for people will just result in what bought "optimized" setups usually result in - confusion for people who suddenly discover an issue due to a config option they don't know about and they can't even tell how to fix that, because they never took time to understand what they're doing.

**Reply by bagdydad** [reply]
Actually, that's not exactly what I wanted to say. You are right about what you said and the reasons you stated. Someone who does not know what these settings do will not know what they do unless they read the description, even if they apply them manually. This tool allows you to apply settings you already know to multiple servers, or if you're using your server in experimental settings like me and need to constantly reset the files, it allows you to take quick action without having to make the same settings from scratch every time.

**Reply by YouHaveTrouble** [reply]
In that case I can argue that the naming of the repo/app is disingenuous at best. It just sets what most people would define as random values. It would be at least useful if it read yaml files in the scripts directory and copy pasted the values from there to multiple servers. I would not call that "optimizer" in any case.

**Reply by bagdydad** [reply]
Thanks for response.
Let me start with the insincerity of the application name. The purpose of the optimizer word in the application name is to automatically apply the general settings that an average server can use in terms of sacrifice/benefit ratio to the server. In other words, it does what it says. If your server needs more specific configurations, you can still add the relevant settings to the application and automatically apply them to your servers. This is not a clickbait or misleading repository name. I also don't understand why most people would define the settings accepted by the community as random values ​​according to your statement? You said it would be useful if they read the files in the directory and copied the values; it is no different than copying the files directly. The difference of this tool is that it allows you to change only the necessary settings without disturbing your current configuration. I would call it "optimizer" in any case.
Kind regards.

**Reply by YouHaveTrouble** [reply]
> general settings that an average server can use in terms of sacrifice/benefit ratio to the server.

Settings like that do not exist. Every server is different and every server owner has their own limits on how much they're willing to sacrifice in terms of mechanics and behavior.

The entire concept of your script is setting values that, let's not lie to ourselves, will never be looked at or changed by the user. They will just see the word "optimizer" and run it. It does not matter how bold or big you make the font saying that you're not responsible for any issues. you're giving way too much credit to the average person.

The only scenario where something like this would be acceptable is to have default server configs in the location of the script and make the script copy over the changed values. Changing users configuration without them having to change the config themselves is going to be hurting them in the long run, because they're headed to the scenario I described before - not even knowing what settings were changed or why. They need to change them fully knowing what they're changing.

**Reply by bagdydad** [reply]
> Settings like that do not exist. Every server is different and every server owner has their own limits on how much they're willing to sacrifice in terms of mechanics and behavior.

I already understand the subject you are trying to convey in this discussion about "a man stuck in time". Instead of trying to understand what I am trying to convey to you, you are adopting a single thought, as in the expression "putting on horse blinkers", and saying the same things. I think I am wasting my time by explaining your thoughts about me deceiving myself, which is not the case.

> The entire concept of your script is setting values that, let's not lie to ourselves, will never be looked at or changed by the user.

Who told you this? I don't. It's just an empty thought that you've made up in your own head.

> They will just see the word "optimizer" and run it. It does not matter how bold or big you make the font saying that you're not responsible for any issues. you're giving way too much credit to the average person.

Even so, the settings do not involve any special sacrifices, and there is almost no chance of an unwanted change on the server. Even if someone who knows nothing downloads and runs it just because it says "optimizer", there will be no loss just it improves the server.

(Optional: If something is big and thick, you have to be careful with it.)

> The only scenario where something like this would be acceptable is to have default server configs in the location of the script and make the script copy over the changed values. Changing users configuration without them having to change the config themselves is going to be hurting them in the long run, because they're headed to the scenario I described before - not even knowing what settings were changed or why. They need to change them fully knowing what they're changing.

Just like the scenario you suggested for my application, here too they may only look at which file they need to change which setting and may not care about what the setting does, just because it says "optimization".

Thanks for your advice. I really liked your advice and will update the app accordingly, so the settings are easier to read and change, I just wanted to keep it simple.
